core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -443,4 +446,52 @@ public StructuralLoadFilter parseStructuralLoad(Element el) throws InvalidXMLExc
   public TimeFilter parseTimeFilter(Element el) throws InvalidXMLException {
     return new TimeFilter(XMLUtils.parseDuration(el, null));
   }
+
+  @MethodParser("match-state")
+  public MatchStateFilter parseMatchState(Element el) throws InvalidXMLException {
+    return parseMatchStateFilter(el.getValue(), el);
+  }
+
+  @MethodParser("match-started")
+  public MatchStateFilter parseMatchStarted(Element el) throws InvalidXMLException {
+    return parseMatchStateFilter("started", el);
+  }
+
+  @MethodParser("match-running")
+  public MatchStateFilter parseMatchRunning(Element el) throws InvalidXMLException {
+    return parseMatchStateFilter("running", el);
+  }
+
+  @MethodParser("match-finished")
+  public MatchStateFilter parseMatchFinished(Element el) throws InvalidXMLException {
+    return parseMatchStateFilter("finished", el);
+  }
+
+  public MatchStateFilter parseMatchStateFilter(String matchState, Element el)
+      throws InvalidXMLException {
+
+    Predicate<Match> matchPredicate = null;
+
+    switch (matchState) {
+      case "running":
+        matchPredicate = m -> m.getPhase() == MatchPhase.RUNNING;
+        break;
+      case "finished":
+        matchPredicate = m -> m.getPhase() == MatchPhase.FINISHED;
+        break;
+      case "starting":
+        matchPredicate = m -> m.getPhase() == MatchPhase.STARTING;
+        break;
+      case "before":
+        matchPredicate = m -> m.getPhase() == MatchPhase.IDLE;
+        break;
+      case "started":
+        matchPredicate =
+            m -> m.getPhase() == MatchPhase.RUNNING || m.getPhase() == MatchPhase.FINISHED;
+    }
+    if (matchPredicate == null)
+      throw new InvalidXMLException("Invalid or no match state found", el);
```

Comment:
Seems like this should be a private function `parseMatchPhase(String) -> MatchPhase`